allennlp/data/dataset_readers/penn_tree_bank.py

```diff
@@ -196,7 +200,18 @@ def _get_gold_spans(self, # pylint: disable=arguments-differ
                 child_start = end
             # Set the end index of the current span to
             # the last appended index - 1, as the span is inclusive.
-            typed_spans[(index, end - 1)] = tree.label()
+            span = (index, end - 1)
+            current_span_label = typed_spans.get(span)
+            if current_span_label is None:
+                # This span doesn't have nested labels, just
+                # use the current node's label.
+                typed_spans[span] = tree.label()
+            else:
+                # This span has already been added, so prepend
+                # this label (as we are traversing the tree from
+                # the bottom up).
+                typed_spans[span] = tree.label() + "-" + current_span_label
```

Comment:
That's true, but also the case for any other model that we have? I'm not sure it's a huge problem as the vocab will be constructed from the train, dev and test splits. I guess the solution would have the same effect - have a long list of enumerated pairs of labels and raise an error if you encounter one here, rather than just crashing later on in the indexing step.

Reply:
(this also wouldn't crash at "true" test time because this is for labels, not inputs)

Reply:
The trouble is that if you see a new combination at test time (and forgot to include the test vocab, or are running on new data), you won't even be able to evaluate the model, because there's no mechanism for saying that this new combination is unknown.  And yeah, that's also true for a lot of other models, it just seems more likely to happen here, because you're sampling from the cross product of possible nonterminals.  Not a huge deal to figure out how to handle this later.

Reply:
Didn't see your second note when I wrote the above - the whole reason I'm pointing this out is that it _will_ crash at test time, if you're using `evaluate`, because indexing the test inputs will crash.  But anyway, yeah, a TODO is fine.  Handle this when it actually becomes a problem.